### PR TITLE
added counters in common for proposals, discussions, and messages.

### LINKED ISSUE #1307: Common Card - Add number of Discussions, Proposals and Messages to daos collection item
We have this new design. And now we should display numbers of proposals, discussions and messages before opening the common page. If we want to keep a good app performance, we should implement it on the backend side. Kinda the same as we did for number of messages for discussions.
Related to https://github.com/daostack/common-backend/issues/1281.
[https://www.figma.com/file/9fox8n9BHoqMSTzttXvkWe/🟩-Dev---Common-v1.0---Mobile-App?node-id=932%3A25546](https://www.figma.com/file/9fox8n9BHoqMSTzttXvkWe/%F0%9F%9F%A9-Dev---Common-v1.0---Mobile-App?node-id=932%3A25546)

### PR DESCRIPTION
**Pull request by noamso1**

### CODE REVIEW

**Comment by noamso1** [comment]
Done - now you're gonna have:
common.proposalCount (only funding proposals are counted)
common.discussionCount
common.messageCount

* note - the migration should run initial counting. not included in this PR.

File: functions/src/proposals/triggers/updateCommonProposalCount.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { ICommon } from '../../shared/interfaces/commons/ICommon';
+import { commonsDb } from '../../commons/db'
+
+export async function countProposals(commonId: string, v: number) {
+    try {
+        if ( commonId ) {
+            let common: ICommon = await commonsDb.get( commonId )
```
Comment (ariel-dao) [inline functions/src/proposals/triggers/updateCommonProposalCount.ts:7]: use const

File: functions/src/proposals/triggers/updateCommonProposalCount.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+            let x : number = common.proposalCount; if ( !x ) x = 0
```
Comment (ariel-dao) [inline functions/src/proposals/triggers/updateCommonProposalCount.ts:8]: use const

File: functions/src/proposals/triggers/updateCommonProposalCount.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+            common.proposalCount = x + v
+            await commonsDb.update( common )
+        }
+    } catch {
+        logger.error("error updating common.discussionCount");
```
Comment (ariel-dao) [inline functions/src/proposals/triggers/updateCommonProposalCount.ts]: more details on error

File: functions/src/discussionMessage/triggers.ts [hunk 4]
```diff
@@ -28,12 +30,24 @@ exports.watchForDeletedMessages = functions.firestore
   });
 
 async function countDiscussionMessages(discussionId: string, v: number) {
+  let commonId : string
   try {
-    let discussion: IDiscussion = await discussionDb.getDiscussion(discussionId)
-    let x = discussion.messageCount; if (!x) x = 0
+    let discussion: IDiscussion = await discussionDb.getDiscussion( discussionId )
```
Comment (ariel-dao) [inline functions/src/discussionMessage/triggers.ts:36]: const

File: functions/src/discussionMessage/triggers.ts [hunk 5]
```diff
@@ -28,12 +30,24 @@ exports.watchForDeletedMessages = functions.firestore
...
+    let x : number = discussion.messageCount; if ( !x ) x = 0
```
Comment (ariel-dao) [inline functions/src/discussionMessage/triggers.ts:37]: const

File: functions/src/discussionMessage/triggers.ts [hunk 6]
```diff
@@ -28,12 +30,24 @@ exports.watchForDeletedMessages = functions.firestore
...
+    commonId = discussion.commonId
     discussion.messageCount = x + v
-    await discussionDb.updateDiscussion(discussionId, discussion)
+    await discussionDb.updateDiscussion( discussionId, discussion )
   } catch {
     logger.error("No discussionId");
```
Comment (ariel-dao) [inline functions/src/discussionMessage/triggers.ts]: more details on error

File: functions/src/discussionMessage/triggers.ts [hunk 7]
```diff
@@ -28,12 +30,24 @@ exports.watchForDeletedMessages = functions.firestore
...
   }
+  if ( commonId ) {
+    try {
+      let common: ICommon = await commonsDb.get( commonId )
+      let x : number = common.messageCount; if ( !x ) x = 0
+      common.messageCount = x + v
+      await commonsDb.update( common )
+    } catch {
+      logger.error("error updating common.messageCount");
```
Comment (ariel-dao) [inline functions/src/discussionMessage/triggers.ts]: details

File: functions/src/discussion/triggers.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+async function countDiscussions(commonId: string, v: number) {
+    try {
+        if ( commonId ) {
+            let common: ICommon = await commonsDb.get( commonId )
+            let x : number = common.discussionCount; if ( !x ) x = 0
+            common.discussionCount = x + v
+            await commonsDb.update( common )
+        }
+    } catch {
+        logger.error("error updating common.discussionCount");
```
Comment (ariel-dao) [inline functions/src/discussion/triggers.ts]: details

File: functions/src/discussion/triggers.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  .document(`/${COLLECTIONS.Discussion}/{id}`)
+  .onDelete(async (snap) => {
+    const discussion = snap.data() as IDiscussion;
+    await countDiscussions(discussion.commonId, -1);
+  });
+
+async function countDiscussions(commonId: string, v: number) {
+    try {
+        if ( commonId ) {
+            let common: ICommon = await commonsDb.get( commonId )
```
Comment (ariel-dao) [inline functions/src/discussion/triggers.ts]: use const

**Review by ariel-dao — CHANGES_REQUESTED** [review]
please review and change

File: functions/src/proposals/triggers/updateCommonProposalCount.ts (hunk 2 above)
Comment (noamso1) [reply functions/src/proposals/triggers/updateCommonProposalCount.ts:8]: both "common" and "x" are not constants

File: functions/src/discussionMessage/triggers.ts (hunk 4 above)
Comment (noamso1) [reply functions/src/discussionMessage/triggers.ts:36]: they are not constants

**Review by ariel-dao — APPROVED** [review]

**Pull request merged**